packages/ui/src/helpers/authenticator/__tests__/listenToAuthHub.test.ts

```diff
@@ -75,6 +77,30 @@ describe('listenToAuthHub', () => {
     Hub.dispatch('auth', { event: 'tokenRefresh_failure' });
     expect(sendSpy).not.toHaveBeenCalled();
 
+    unsubscribe();
+  });
+  it('ignores autoSignIn event when state is authenticated', () => {
+    const unsubscribe = listenToAuthHub(authenticatedStateMachine as any);
+
+    Hub.dispatch('auth', { event: 'autoSignIn' });
+    expect(unAuthSendSpy).not.toBeCalledWith();
+
+    unsubscribe();
+  });
+  it('responds to autoSignIn event when state is unauthenticated', () => {
```

Comment:
NIT: let's group by use cases, and put happy cases above unhappy cases.
(1) responds to autoSignIn event when state is unauthenticated
(2) ignores autoSignIn event when state is authenticated
(3) responds to autoSignIn_failure event

Reply:
Ok I moved it around